src/GitLinkTask/LinkProject.cs

```diff
@@ -7,22 +7,63 @@
 
 namespace GitLinkTask
 {
+    using System;
     using System.IO;
+    using System.Linq;
+    using GitLink;
+    using GitLink.Pdb;
+    using GitLink.Providers;
+    using Catel.Logging;
     using Microsoft.Build.Framework;
     using Microsoft.Build.Utilities;
 
-    public class LinkProject : ToolTask
+    public class LinkProject : Task
     {
         [Required]
-        public ITaskItem SolutionDirectory { get; set; }
+        public ITaskItem PdbFile { get; set; }
 
         [Required]
-        public string ProjectName { get; set; }
+        public ITaskItem[] SourceFiles { get; set; }
 
-        protected override string ToolName => "GitLink.exe";
+        public bool DownloadWithPowershell { get; set; }
 
-        protected override string GenerateFullPathToTool() => Path.Combine(Path.GetDirectoryName(GetType().Assembly.Location), "GitLink.exe");
+        public bool SkipVerify { get; set; }
 
-        protected override string GenerateCommandLineCommands() => $"{SolutionDirectory.FullPath()} -include {ProjectName}";
+        public string GitRemoteUrl { get; set; }
```

Comment:
I think we would also need the branch + sha of the commit in case we are not building against a local .git directory.